Source/gendung.cpp

```diff
@@ -96,8 +96,7 @@ bool WillThemeRoomFit(int floor, int x, int y, int minSize, int maxSize, int *wi
 	int xCount = 0;
 	int yCount = 0;
 
-	// BUGFIX: change '&&' to '||' (fixed)
-	if (x > DMAXX - maxSize || y > DMAXY - maxSize) {
+	if ((x + maxSize >= DMAXX || y + minSize >= DMAXY) && (x + minSize >= DMAXX && y + maxSize >= DMAXY)) {

```

Comment:
why the mixed check?

Reply:
To get as close as possible to handeling all OOB issues

Reply:
This one confused me too. Having a comment in the code with your explanation from https://github.com/diasurgical/devilutionX/pull/2356#discussion_r667533450 would be great

Reply:
yes but wasn't it supposed to be minSize x2 in the first check then maxSize x2 in the second one? I don't understand the current one

Reply:
Hmm I thought it required one of the sides to be maxSize, but looking at this again it might be ok with minSize x minSize rooms. The original would actually reject a bunch of valid rooms if it's in the corner so we might need to keep that.